src/vs/platform/native/electron-main/nativeHostMainService.ts

```diff
@@ -742,14 +757,315 @@ export class NativeHostMainService extends Disposable implements INativeHostMain
+		const scaledBounds = {
+			x: (nodeData.bounds.x + offsetX) * zoomFactor,
+			y: (nodeData.bounds.y + offsetY) * zoomFactor,
+			width: nodeData.bounds.width * zoomFactor,
+			height: nodeData.bounds.height * zoomFactor
+		};
+
+		return { outerHTML: nodeData.outerHTML, computedStyle: nodeData.computedStyle, bounds: scaledBounds };
+	}
+
+	async getNodeData(sessionId: number, debuggers: any, window: BrowserWindow): Promise<NodeDataResponse> {
+		return new Promise((resolve) => {
+			const onMessage = async (event: any, method: string, params: { backendNodeId: number }) => {
+				if (method === 'Overlay.inspectNodeRequested') {
+
+					await debuggers.sendCommand('Runtime.evaluate', {
+						expression: `(() => {
+								const style = document.getElementById('__pseudoBlocker__');
+								if (style) style.remove();
+							})();`,
+					}, sessionId);
+
+
+					debuggers.off('message', onMessage);
+					const backendNodeId = params?.backendNodeId;
+					if (!backendNodeId) {
+						throw new Error('Missing backendNodeId in inspectNodeRequested event');
+					}
+
+					try {
+						await debuggers.sendCommand('DOM.getDocument', {}, sessionId);
+						const { nodeIds } = await debuggers.sendCommand('DOM.pushNodesByBackendIdsToFrontend', { backendNodeIds: [backendNodeId] }, sessionId);
+						if (!nodeIds || nodeIds.length === 0) {
+							throw new Error('Failed to get node IDs.');
+						}
+						const nodeId = nodeIds[0];
+
+						const { model } = await debuggers.sendCommand('DOM.getBoxModel', { nodeId }, sessionId);
+						if (!model) {
+							throw new Error('Failed to get box model.');
+						}
+
+						const margin = model.margin;
+						const x = margin[0];
+						const y = margin[1] + 32.4 + 35; // 32.4 is height of the title bar, 35 is height of the tab bar
+						const width = margin[2] - margin[0];
+						const height = margin[5] - margin[1];
+
+						const matched = await debuggers.sendCommand('CSS.getMatchedStylesForNode', { nodeId }, sessionId);
+						if (!matched) {
+							throw new Error('Failed to get matched css.');
+						}
+
+						const response: ComputedStyleResponse = await debuggers.sendCommand('CSS.getComputedStyleForNode', { nodeId }, sessionId);
+						if (!response || !response.computedStyle) {
+							throw new Error('Failed to get computed style.');
+						}
+
+						const formatted = this.formatMatchedStyles(matched, response);
+
+						const { outerHTML } = await debuggers.sendCommand('DOM.getOuterHTML', { nodeId }, sessionId);
+						if (!outerHTML) {
+							throw new Error('Failed to get outerHTML.');
+						}
+
+						resolve({
+							outerHTML,
+							computedStyle: formatted,
+							bounds: { x, y, width, height }
+						});
+					} catch (err) {
+						debuggers.detach();
+						throw new Error(`Failed to get node data: ${err}`);
+					}
+				}
+			};
+
+			window.webContents.on('ipc-message', async (event, channel) => {
+				if (channel === 'vscode:cancelElementSelection') {
```

Comment:
Prefer to use the existing IPC channels (like a `cancelGetElementData`) instead of one-off messages like this.